src/awst_build/eb/storage/box/box.ts

```diff
@@ -62,6 +67,55 @@ export class BoxExpressionBuilder extends BoxProxyExpressionBuilder<BoxPType> {
   }
 }
 
+class BoxCreateFunctionBuilder extends FunctionBuilder {
+  constructor(
+    private boxValue: BoxValueExpression,
+    private contentType: PType,
+    sourceLocation: SourceLocation,
+  ) {
+    super(sourceLocation)
+  }
+
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const {
+      args: [{ size }],
+    } = parseFunctionArgs({
+      args,
+      typeArgs,
+      genericTypeArgs: 0,
+      callLocation: sourceLocation,
+      funcName: 'Box.create',
+      argSpec: (a) => [a.obj({ size: a.optional(uint64PType) })],
+    })
+    if (size) {
+      return instanceEb(
+        nodeFactory.intrinsicCall({
+          opCode: 'box_create',
+          stackArgs: [this.boxValue.key, size.resolve()],
+          wtype: wtypes.boolWType,
+          immediates: [],
+          sourceLocation,
+        }),
+        boolPType,
+      )
+    }
+    const minBoxSize = getMinBoxSize(this.contentType)
+    if (minBoxSize === 0n) {
+      logger.warn(sourceLocation, `${this.contentType} does not have a minimum byte size. Box will be created with length of 0`)
```

Comment:
Maybe this should only work with fixed size items? Otherwise the content could be an invalid value.